arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -649,4 +648,111 @@ fvm_size_type fvm_lowered_cell_impl<B>::fvm_intdom(
     return intdom_id;
 }
 
+template <typename B>
+fvm_probe_info fvm_lowered_cell_impl<B>::resolve_probe_address(
+    std::size_t cell_idx,
+    const util::any& paddr,
+    const fvm_cv_discretization& D,
+    const fvm_mechanism_data& M,
+    const std::vector<target_handle>& handles,
+    const std::unordered_map<std::string, mechanism*>& mech_tbl)
+{
+    using util::any_cast;
+    using util::value_by_key;
+    using util::binary_search_index;
+    using namespace cv_prefer;
+
+    // Probe address can be one of a number of cable cell-specific
+    // probe types; dispatch on type and compute probe info accordingly.
+    //
+    // Mechanisms or ions not being instantiated at a particular address
+    // is not treated as an error; a request for a state variable that
+    // does not exist on a mechanism is, however.
+
+    auto location_cv = [&](auto* p, cv_prefer::type prefer) -> fvm_index_type {
+        return D.geometry.location_cv(cell_idx, p->location, prefer);
+    };
+
+    struct mechanism_data {
+        const fvm_value_type* state_data = nullptr;
+        unsigned id;
+        explicit operator bool() const { return state_data; }
+    };
+
+    auto lookup_mechanism_data = [&](auto* p) -> mechanism_data {
+        if (mechanism* m = value_by_key(mech_tbl, p->mechanism).value_or(nullptr)) {
+            const fvm_value_type* data = B::mechanism_field_data(m, p->state);
```

Comment:
What is `B`? This is too far from the definition to make for a really useful single letter name.

Reply:
All the other member functions defined around here also use `B` for the type parameter. Change them all to `Backend`?